### ISSUE TITLE
unable to change fqdn pool member

### ISSUE THREAD

**Issue by chen23** [F5Networks/f5-appsvcs-extension#46]
##### ISSUE TYPE
 - Bug Report

##### AS3 BUILD/ VERSION
```
f5-appsvcs-3.7.0-7.noarch
```

##### BIGIP VERSION

```
Sys::Version
Main Package
  Product     BIG-IP
  Version     13.1.1
  Build       0.0.4
  Edition     Final
  Date        Fri Jul 20 17:55:49 PDT 2018
```

##### SUMMARY

unable to change a fqdn pool member

##### STEPS TO REPRODUCE

original payload
```
{
  "class": "ADC",
  "schemaVersion": "3.0.0",
  "label": "autoscale_waf",
  "id": "AUTOSCALE_WAF",
  "remark": "Autoscale WAF",
  "waf": {
    "class": "Tenant",
    "Shared": {
      "class": "Application",
      "template": "shared",
      "serviceAddress": {
        "class": "Service_Address",
        "virtualAddress": "0.0.0.0"
      },
      "policyWAF": {
        "class": "WAF_Policy",
        "file": "/config/cloud/asm-policy-linux-high.xml"
      }
    },
    "http": {
      "class": "Application",
      "template": "http",
      "serviceMain": {
        "class": "Service_HTTP",
        "virtualAddresses": [
          {
            "use": "/waf/Shared/serviceAddress"
          }
        ],
        "serverTLS": {
          "bigip": "/Common/example-clientssl-profile"
        },
        "snat": "auto",
        "securityLogProfiles": [
          {
            "bigip": "/Common/Log illegal requests"
          }
        ],
        "pool": "pool",
        "policyWAF": {
          "use": "/waf/Shared/policyWAF"
        },
        "virtualPort": 443
      },
      "pool": {
        "class": "Pool",
        "monitors": [
          "http"
        ],
        "members": [
          {
            "autoPopulate": true,
            "hostname": "www.example.com",
            "servicePort": 80,
            "addressDiscovery": "fqdn"
          }
        ]
      }
    }
  }
}
```
new payload:
```
{
  "class": "ADC",
  "schemaVersion": "3.0.0",
  "label": "autoscale_waf",
  "id": "AUTOSCALE_WAF",
  "remark": "Autoscale WAF",
  "waf": {
    "class": "Tenant",
    "Shared": {
      "class": "Application",
      "template": "shared",
      "serviceAddress": {
        "class": "Service_Address",
        "virtualAddress": "0.0.0.0"
      },
      "policyWAF": {
        "class": "WAF_Policy",
        "file": "/config/cloud/asm-policy-linux-high.xml"
      }
    },
    "http": {
      "class": "Application",
      "template": "https",
      "serviceMain": {
        "class": "Service_HTTPS",
        "virtualAddresses": [
          {
            "use": "/waf/Shared/serviceAddress"
          }
        ],
        "serverTLS": {
          "bigip": "/Common/example-clientssl-profile"
        },
        "snat": "auto",
        "securityLogProfiles": [
          {
            "bigip": "/Common/Log illegal requests"
          }
        ],
        "pool": "pool",
        "policyWAF": {
          "use": "/waf/Shared/policyWAF"
        },
        "virtualPort": 443
      },
      "pool": {
        "class": "Pool",
        "monitors": [
          "http"
        ],
        "members": [
          {
            "autoPopulate": true,
            "hostname": "ip-10-1-10-188.ec2.internal",
            "servicePort": 80,
            "addressDiscovery": "fqdn"
          }
        ]
      }
    }
  }
}
```
##### EXPECTED RESULTS

pool updated

##### ACTUAL RESULTS

```
{
  "results": [
    {
      "message": "declaration failed",
      "response": "01070110:3: Node address '/Common/_auto_93.184.216.34' is referenced by a member of pool '/waf/http/pool'.",
      "code": 422,
      "host": "localhost",
      "tenant": "waf",
      "runTime": 9509
    }
  ],
  "declaration": {
    "waf": {
      "class": "Tenant",
      "Shared": {
        "class": "Application",
        "template": "shared",
        "serviceAddress": {
          "class": "Service_Address",
          "virtualAddress": "0.0.0.0"
        },
        "policyWAF": {
          "class": "WAF_Policy",
          "file": "/config/cloud/asm-policy-linux-high.xml"
        }
      },
      "http": {
        "class": "Application",
        "template": "https",
        "serviceMain": {
          "class": "Service_HTTPS",
          "virtualAddresses": [
            {
              "use": "/waf/Shared/serviceAddress"
            }
          ],
          "serverTLS": {
            "bigip": "/Common/example-clientssl-profile"
          },
          "snat": "auto",
          "securityLogProfiles": [
            {
              "bigip": "/Common/Log illegal requests"
            }
          ],
          "pool": "pool",
          "policyWAF": {
            "use": "/waf/Shared/policyWAF"
          },
          "virtualPort": 443
        },
        "pool": {
          "class": "Pool",
          "monitors": [
            "http"
          ],
          "members": [
            {
              "autoPopulate": true,
              "hostname": "www.example.com",
              "servicePort": 80,
              "addressDiscovery": "fqdn"
            }
          ]
        }
      }
    },
    "class": "ADC",
    "schemaVersion": "3.0.0",
    "label": "autoscale_waf",
    "id": "AUTOSCALE_WAF",
    "remark": "Autoscale WAF",
    "updateMode": "selective",
    "controls": {
      "archiveTimestamp": "2018-12-19T04:07:54.679Z"
    }
  },
  "code": 422
}
```

**Comment by lephubui** [comment]
Hi @chen23 Thanks for reporting, and we have opened an internal issue for it.

**Comment by kaustriaf5** [comment]
Slated to be fixed in 3.8 release

**Comment by kaustriaf5** [comment]
The fix for this is now available in [3.8](https://github.com/F5Networks/f5-appsvcs-extension/releases/tag/v3.8.0)
Closing this issue.
